pkg/controllers/clusterresourcebindingwatcher/watcher.go

```diff
@@ -0,0 +1,119 @@
+			"CRPTrackingLabel is not present",
+			"clusterResourceBinding", bindingRef)
+		// This is not a situation that the controller can recover by itself. Should the label
+		// value be corrected, the controller will be triggered again.
+		return ctrl.Result{}, nil
+	}
+
+	// Enqueue the CRP name for reconciling.
+	r.PlacementController.Enqueue(crpName)
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	customPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			// Ignore creation events.
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// Ignore deletion events.
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// Check if the update event is valid.
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				err := controller.NewUnexpectedBehaviorError(fmt.Errorf("update event is invalid"))
+				klog.ErrorS(err, "Failed to process update event")
+				return false
+			}
+			oldBinding, oldOk := e.ObjectOld.(*fleetv1beta1.ClusterResourceBinding)
+			newBinding, newOk := e.ObjectNew.(*fleetv1beta1.ClusterResourceBinding)
+			if !oldOk || !newOk {
+				err := controller.NewUnexpectedBehaviorError(fmt.Errorf("failed to cast runtime objects in update event to cluster resource binding objects"))
+				klog.ErrorS(err, "Failed to process update event")
+				return false
+			}
+
+			return isGenerationUpdated(oldBinding, newBinding) ||
+				isConditionUpdated(oldBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted)), newBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted))) ||
+				isConditionUpdated(oldBinding.GetCondition(string(fleetv1beta1.ResourceBindingOverridden)), newBinding.GetCondition(string(fleetv1beta1.ResourceBindingOverridden))) ||
+				isConditionUpdated(oldBinding.GetCondition(string(fleetv1beta1.ResourceBindingWorkCreated)), newBinding.GetCondition(string(fleetv1beta1.ResourceBindingWorkCreated))) ||
+				isConditionUpdated(oldBinding.GetCondition(string(fleetv1beta1.ResourceBindingApplied)), newBinding.GetCondition(string(fleetv1beta1.ResourceBindingApplied))) ||
+				isConditionUpdated(oldBinding.GetCondition(string(fleetv1beta1.ResourceBindingAvailable)), newBinding.GetCondition(string(fleetv1beta1.ResourceBindingAvailable)))
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&fleetv1beta1.ClusterResourceBinding{}).
+		WithEventFilter(customPredicate).
+		Complete(r)
+}
+
+func isGenerationUpdated(oldBinding, newBinding *fleetv1beta1.ClusterResourceBinding) bool {
+	return oldBinding.Generation != newBinding.Generation
+}
+
+func isConditionUpdated(oldCondition, newCondition *metav1.Condition) bool {
+	if oldCondition == nil && newCondition == nil {
+		return false
+	}
+	if oldCondition == nil || newCondition == nil {
+		return true
+	}
+	return oldCondition.Status != newCondition.Status
```

Comment:
```suggestion
	return oldBinding.ObservedGeneration != oldCondition.ObservedGeneration != newCondition.ObservedGeneration || oldCondition.Status != newCondition.Status || oldCondition.Reason != newCondition.Reason || 
```

usually the reason won't be changed if the status & observedGeneration have not been changed. Just in case.